File: src/test/java/com/gitlab/daring/image/util/EnumUtilsTest.java
package com.gitlab.daring.image.util;

import org.junit.Test;

import static com.gitlab.daring.image.util.EnumUtils.findEnum;
import static com.gitlab.daring.image.util.EnumUtilsTest.TestEnum.*;
import static org.junit.Assert.assertEquals;

public class EnumUtilsTest {

	@Test
	public void testFindEnum() {
		checkFindEnum(One, "One");
		checkFindEnum(One, "one");
		checkFindEnum(One, "o");
		checkFindEnum(Two, "two");
		checkFindEnum(Two, "t");
		checkFindEnum(Three, "th");
	}

	void checkFindEnum(TestEnum exp, String name) {
		assertEquals(exp, findEnum(TestEnum.values(), name));
	}

	enum TestEnum { One, Two, Three }

}

File: src/test/kotlin/com/gitlab/daring/image/util/EnumUtilsTest.kt
package com.gitlab.daring.image.util

import com.gitlab.daring.image.util.EnumUtils.findEnum
import com.gitlab.daring.image.util.EnumUtilsTest.TestEnum.*
import org.junit.Assert.assertEquals
import org.junit.Test

class EnumUtilsTest {

    @Test
    fun testFindEnum() {
        checkFindEnum(One, "One")
        checkFindEnum(One, "one")
        checkFindEnum(One, "o")
        checkFindEnum(Two, "two")
        checkFindEnum(Two, "t")
        checkFindEnum(Three, "th")
    }

    fun checkFindEnum(exp: TestEnum, name: String) {
        assertEquals(exp, findEnum(TestEnum::class.java, name))
    }

    enum class TestEnum { One, Two, Three }

}